### DISCUSSION TITLE
Force prisma order of columns

### DISCUSSION THREAD

**Discussion by robokonk** [prisma/prisma#18470] [Q&A]
I have a problem where the order of columns in the table is often different from the order in schema.prisma. This issue occurs when I create a table from scratch, not when I add a new column to the end of the table.

How can I enforce the desired column order? Currently, my only idea is to create the table manually using SQL. Are there any other suggestions?

**Comment by nurul3101** [comment]
Hello @robokonk 👋 

You can do [introspection](https://www.prisma.io/docs/concepts/components/introspection) to generate the schema file from your database. After introspection, the order of columns in the database and in the schema file should be the same. 

So you can create a table in database through SQL and then run `npx prisma db pull` to get this new table information in the schema file.

**Reply by nurul3101** [reply]
@robokonk Did you had a chance to try out introspection? Do let us know in case you face any issues.

**Reply by nurul3101** [reply]
@robokonk It would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 

Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏
